US Congress (Independent Publisher)

krautrocker opened the pull request:
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [x] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [x] I attest that the connector works and I verified by deploying and testing all the operations. 
- [x] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [x] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [x] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [x] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [x] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [x] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [x] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [x] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

krautrocker committed:
- Initial commit

krautrocker commented:
Proof of life:

Test screenshots:

![Screenshot 2025-02-11 230321](https://github.com/user-attachments/assets/62c8ffdf-1c4f-494c-8de0-f5cb45628c4f)

![Screenshot 2025-02-11 225642](https://github.com/user-attachments/assets/5a9499be-ff36-4cfd-ab79-deecbf0a6aee)
![Screenshot 2025-02-11 225513](https://github.com/user-attachments/assets/876aea3e-052a-4166-9715-1e115682ae92)
![Screenshot 2025-02-11 225810](https://github.com/user-attachments/assets/9a7b0f58-804b-4a5d-ae83-680d3f3f8997)

Operations working within a flow:

![Screenshot 2025-02-11 230931](https://github.com/user-attachments/assets/d743ffb7-d7df-4d35-afb1-db1374d25920)
![Screenshot 2025-02-11 230922](https://github.com/user-attachments/assets/426a7540-044d-44dc-ae42-445688c2b1f5)
![Screenshot 2025-02-11 230937](https://github.com/user-attachments/assets/3654db73-1b9d-4eb9-b63f-613b85f813e7)

krautrocker committed:
- Modified x-ms-summary property for endpoint parameters

- paconn validated, no errors.

krautrocker commented:
Work to to do:

1. Update x-ms-summary property for endpoints
2. Add response schemas to endpoints

krautrocker committed:
- Modified response schema for endpoints.

krautrocker commented:
Test screenshots and proof of life.

Test environment:

![Screenshot 2025-02-11 225642](https://github.com/user-attachments/assets/4cc62380-2a67-486b-841f-cbf1789403c3)
![Screenshot 2025-02-11 225513](https://github.com/user-attachments/assets/194de804-408b-4929-b17f-8fc5d7fdebd2)
![Screenshot 2025-02-11 225810](https://github.com/user-attachments/assets/a23011fe-00f7-4580-8531-40356b2d4db8)

Working within a flow:

![Screenshot 2025-02-11 230937](https://github.com/user-attachments/assets/f1a421fc-0331-486b-956a-1a17c7bbaaec)
![Screenshot 2025-02-11 230931](https://github.com/user-attachments/assets/0b5f8b56-84e1-4380-9ef0-2961f183ee7d)
![Screenshot 2025-02-11 230922](https://github.com/user-attachments/assets/c414b807-a422-4f54-9652-6a9442f21312)

krautrocker committed:
- Added response schemas for the amendments endpoints

- paconn validated, no errors

krautrocker committed:
- Added response schemas to the bills, laws and summaries endpoints.

krautrocker committed:
- Added response schemas to Congress endpoints

krautrocker committed:
- Added response schemas to Members endpoints

- paconn validated, no errors

krautrocker committed:
- Added response schemas to the Committees endpoint

krautrocker committed:
- Added response schema for the Congressional Record endpoint.

- paconn validated, no errors.

krautrocker committed:
- Added response schemas for the Nominations endpoints.

- paconn validated, no warnings

krautrocker committed:
- Added response schemas for Treaty endpoints.

- paconn validated, no errors.

krautrocker committed:
- Added response schemas for Hearing endpoints.

krautrocker committed:
- Re-worded action descriptions

- paconn validated, no errors

krautrocker committed:
- Added response schemas for Daily Congressional Record endpoints.

krautrocker committed:
- Added x-ms-visibility to all endpoints

- paconn validated, no errors

krautrocker committed:
- Added response schemas for Commitee Report endpoints.

- paconn validated, no errors

vmanoharas commented on independent-publisher-connectors/US Congress/apiDefinition.swagger.json:
```diff
@@ -0,0 +1,9694 @@
+  "swagger": "2.0",
+  "info": {
+    "title": "Congressional Research Service",
+    "description": "US Congressional data via Congress.gov",
```
Could you please provide more information in the description?

vmanoharas reviewed (commented):
Hi @krautrocker, 

Thank you for submitting your proposal, we will discuss internallay and get back to you soon. 

However, could you please address one comment?

krautrocker committed:
- Added response schemas for Commitee Print and Committee Meeting endpoints.

- Updated connector description.

- Added privacy policy links to readme.

- paconn validated, no errors.

krautrocker committed:
- Added response schemas to Bound Congressional Record endpoints.

- paconn validated, no errors.

krautrocker committed:
- Added response schemas to House Communication endpoints.

- paconn validated, no errors.

krautrocker committed:
- Added response schemas for House Requirements.

- paconn validated, no errors.

krautrocker committed:
- Added response schemas to Senate Communications endpoints.

- paconn validated, no errors

krautrocker committed:
- Changed operation Id to be consistent with the rest

krautrocker committed:
- Re-did House Communication endpoint, got some errors in the UI while trying to save it

krautrocker committed:
- Updated summaries to include endpoint family names

krautrocker committed:
- Minor clean-up.

- paconn validated, no errors.

krautrocker committed:
- Updated readme

krautrocker commented:
Hi @vmanoharas, is there any update here?

vmanoharas commented:
Hi @krautrocker, 

please allow us time till next week.

vmanoharas commented:
Hi @krautrocker, 

We are in the discussion on this proposal, please allow us some more time. Apologies for the delay.

krautrocker commented:
Sure, please let me know if you have any questions or concerns.

krautrocker commented:
Hi there @vmanoharas - any update here?

vmanoharas commented:
> Hi there @vmanoharas - any update here?

Unfortunately, not yet @krautrocker, we are still working with licensing team.

vmanoharas commented:
Hello @krautrocker, 

Could you please update a single connector per PR? this PR has two connectors. 

Additionally, please remove the proposal word from the title of the PR. 

![image](https://github.com/user-attachments/assets/9e114ab6-fec4-4cd8-9e23-0882bfb35c8d)

krautrocker committed:
Delete independent-publisher-connectors/Bundestag directory

krautrocker commented:
@vmanoharas My bad, should be all set now?

krautrocker committed:
- Shortened summaries

- Removed puncations (periods)

- Added CRS Report endpoints (two total)

- Removed security from each action, as it is defined globally

- paconn validated, no errors and new endpoints confirmed working

vmanoharas reviewed (dismissed):
Dear Partner

I hope you are doing well.
Congratulations, your pull request is approved. We will move forward with the certification process. We are preparing your connector for production deployment and will queue up for the next deployment schedule. Once your connector onboards the next deployment schedule, it starts to deploy your connector in our production environments that typically takes 3-4 weeks.
Please let us know if you have any questions.
Thank you very much for working with us.

vmanoharas commented:
Hi @krautrocker, 

Kindly update the title as per the below point. We are making the following requirement mandatory going forward. 

![image](https://github.com/user-attachments/assets/52c6f3f9-8187-431f-8cfe-12124fe749e9)

krautrocker committed:
- Updated title to include "Independent Publisher"

krautrocker commented:
@vmanoharas Thank you very much as always. Updated the title and will include going forward.

certificationteam commented on independent-publisher-connectors/US Congress/apiProperties.json:
```diff
@@ -0,0 +1,32 @@
+        "title": "Set API Key as Query Parameter",
+        "parameters": {
+          "x-ms-apimTemplateParameter.name": "api_key",
+          "x-ms-apimTemplateParameter.value": "{{api_key}}",
```
Hi @krautrocker, 

We found the error that there is no value passing to this {{api_key}} during our certification process, please correct this.

certificationteam reviewed (commented):
Hi @krautrocker, 

Please address one review comment.

krautrocker committed:
- Removed policyTemplate for properties, not required

- Removed extraneous security setting in path, not required

- paconn validated, no errors

krautrocker commented:
@certificationteam Removed this block from apiProperties. Let me know if that doesn't resolve the issue.

certificationteam reviewed (approved):
Dear Partner

I hope you are doing well.
Congratulations, your pull request is approved. We will move forward with the certification process. We are preparing your connector for production deployment and will queue up for the next deployment schedule. Once your connector onboards the next deployment schedule, it starts to deploy your connector in our production environments that typically takes 3-4 weeks.
Please let us know if you have any questions.
Thank you very much for working with us.

vmanoharas commented:
Hello Partner,
I hope you are doing well.
Congratulations, the deployment of your connector and documentation page was already completed, 
Thank you very much for working with us.
